code/network/multimsgs.cpp

```diff
@@ -8061,11 +8069,21 @@ void process_animation_triggered_packet(ubyte* data, header* hinfo) {
 
 	float delay = time * 0.001f;
 
-	animation::ModelAnimation::s_animationById[animationId]->start(model_get_instance(pmi), direction, forced, instant, pause, &delay);
+	object* objp = multi_get_network_object(netsig);
+
+	auto animation = animation::ModelAnimationSet::s_animationById.find(animationId);
+	if (animation != animation::ModelAnimationSet::s_animationById.end()) {
+		if (special_mode == 0 && objp != nullptr) {
+			animation->second->start(model_get_instance(object_get_model_instance(objp)), direction, forced, instant, pause, &delay);
```

Comment:
`object_get_model_instance()` needs to be separated here for error checking. A return value `-1` is perfectly valid if a ship is joining or respawning since it may not be set up yet, but that triggers an assert in `model_get_instance()`, or it returns null in release builds which crashes in `start()`.

What you could probably do is add a nullptr check to the top of `object_get_model_instance()` returning -1 (it could use it anyway) and use that in place of the `objp != nullptr` test, checking for the `-1` return value instead.